src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -304,6 +306,17 @@ public static IResourceBuilder<PostgresServerResource> WithPgWeb(this IResourceB
                     Directory.CreateDirectory(serverFileMount.Source!);
                 }
 
+                // Need to grant read access to the config file on unix like systems.
+                if (!OperatingSystem.IsWindows())
+                {
+                    // 777
+                    var permissions = UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.UserExecute |
+                                      UnixFileMode.GroupRead | UnixFileMode.GroupWrite | UnixFileMode.GroupExecute |
+                                      UnixFileMode.OtherRead | UnixFileMode.OtherWrite | UnixFileMode.OtherExecute;
+
+                    File.SetUnixFileMode(serverFileMount.Source!, permissions);
+                }
```

Comment:
This can be done as part of the call to CreateDirectory above:

```C#
    public static class Directory
    {
        [UnsupportedOSPlatform("windows")]
        public static DirectoryInfo! CreateDirectory(string! path, UnixFileMode unixCreateMode);
    }
```